test/test.c

```diff
@@ -3001,16 +3004,37 @@ static hb_dict_t * PreparePreset(const char *preset_name)
             return NULL;
         }
     }
-    else
+    else if (preset_import_gui == 1)
     {
         preset = hb_presets_get_default();
+        if (preset == NULL)
+        {
+            fprintf(stderr, "Error loading presets! Aborting.\n");
+            return NULL;
+        }
+        preset = hb_value_dup(preset);
     }
-    if (preset == NULL)
+    else
     {
-        fprintf(stderr, "Error loading presets! Aborting.\n");
-        return NULL;
+        // Default settings
```

Comment:
Initially I agreed, then decided to sleep on it, and have reconsidered.

My concerns with the more explicit approach in this specific case are the inline JSON, increased maintenance for API changes, and way too much code to set a couple params and say "everything else is automatic". Given the above, I believe the current implementation is the simplest, clearest, and most maintainable.